src/data/fetch.ts

```diff
@@ -0,0 +1,44 @@
+import { VectorOperationsApi } from '../pinecone-generated-ts-fetch';
+import type {
+  ResponseError,
+  FetchResponse,
+} from '../pinecone-generated-ts-fetch';
+import {
+  mapHttpStatusError,
+  PineconeConnectionError,
+  extractMessage,
+} from '../errors';
+import { builOptionConfigValidator } from '../validator';
+
+import { Static, Type } from '@sinclair/typebox';
+
+const IdsArray = Type.Array(Type.String({ minLength: 1 }));
+export type IdsArray = Static<typeof IdsArray>;
+
+export const fetch = (api: VectorOperationsApi, namespace: string) => {
+  const validator = builOptionConfigValidator(IdsArray, 'fetch');
+
+  return async (ids: IdsArray): Promise<FetchResponse> => {
+    validator(ids);
```

Comment:
**question** Maybe there are tests for just the validators that already exist outside the context of this PR, but what does the user see when the input fails validation? Is this scenario unit tested elsewhere (asking mostly so I can use it as a doc reference for what the outcome would be since I was curious about fetching without a namespace).

Reply:
Good question. Here's an example failure:

```
> await client.index('jen2').upsert([{ id: 1 }])
Uncaught:
PineconeArgumentError: Argument to upsert has a problem. The property '0' must have required property 'values'. The property '0/id' must be string.
    at /Users/jhamon/workspace/pinecone-ts-client/dist/validator.js:46:15
    at /Users/jhamon/workspace/pinecone-ts-client/dist/validator.js:37:13
    at /Users/jhamon/workspace/pinecone-ts-client/dist/data/upsert.js:62:21
    at step (/Users/jhamon/workspace/pinecone-ts-client/dist/data/upsert.js:33:23)
    at Object.next (/Users/jhamon/workspace/pinecone-ts-client/dist/data/upsert.js:14:53)
    at /Users/jhamon/workspace/pinecone-ts-client/dist/data/upsert.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/jhamon/workspace/pinecone-ts-client/dist/data/upsert.js:4:12)
    at Index.upsert (/Users/jhamon/workspace/pinecone-ts-client/dist/data/upsert.js:57:40)
```

In some earlier functions added to the client library I wrote very extensive unit tests asserting the error message for each type of validation failure, but they are tedious tests to write and brittle if I change the requirements for the params. For these latest actions I skipped over those tests because it has started to feel like repetitive testing of the type checker inside the validator. Since I'm using standardized patterns now for validation and error mapping, the only thing new inside these actions is really defining the types for expected params. I've also been doing a lot of manual testing throughout development to sanity check nothing really wacky is getting shown.